tests/python/unittest/test_arith_deduce_bound.py

```diff
@@ -85,6 +85,44 @@ def test_deduce():
     res3 = tvm.arith.DeduceBound(a, zero <= e3, {b: b_s, c: c_s, d: d_s}, {b: b_s, d: d_s})
     assert str(tvm.ir_pass.Simplify(res3.min_value)) == str(ans3)
 
+    # tests for `EQ` op
+    res4 = tvm.arith.DeduceBound(a, a == b, {}, {})
+    assert_expr_equal(res4.max_value, b)
+    assert_expr_equal(res4.min_value, b)
+
+    # Unsatisfiable `EQ`, variable as one of the Operand
+    res5 = tvm.arith.DeduceBound(a, (a == b), {b: b_s}, {b: b_s})
+    assert str(res5.max_value) == "neg_inf"
+    assert str(res5.min_value) == "pos_inf"
+
+    # variable `a` on the RHS side
+    res6 = tvm.arith.DeduceBound(a, 10 == a, {}, {})
+    assert_expr_equal(res6.max_value, 10)
+    assert_expr_equal(res6.min_value, 10)
+
+    # Add, Sub in `EQ`
+    e4 = ((a - c) == (b + d))
+    ans4 = (b + d + c)
+    res7 = tvm.arith.DeduceBound(a, e4, {b: b_s, c: c_s, d: d_s}, {})
+    assert_expr_equal(res7.max_value, ans4)
+    assert_expr_equal(res7.min_value, ans4)
+
+    # Satisfiable Mul in `EQ` with negative sign
+    res8 = tvm.arith.DeduceBound(a, (5 * a == -10), {}, {})
+    assert_expr_equal(res8.max_value, -2)
+    assert_expr_equal(res8.min_value, -2)
+
+    # Unsatisfiable Mul in `EQ`
+    e5 = (4 * a == b)
+    res9 = tvm.arith.DeduceBound(a, e5, {b: b_s}, {})
+    assert str(res9.max_value) == "neg_inf"
+    assert str(res9.min_value) == "pos_inf"
+
+    # Unsatisfiable Mul in `EQ`
+    res10 = tvm.arith.DeduceBound(a, (b * a == b), {b: b_s}, {})    # simplifier is not able to prove that (b % b == 0)
```

Comment:
@tqchen @xqdan I noticed that TVM wasn't able to prove that `b % b == 0` where `b` is a variable. Is it something expected ?

Reply:
I guess this can be improved @tqchen

Reply:
This can be improved by adding a rewrite pattern